lang/cs/Org.Apache.REEF.Bridge/DriverLauncher.cpp

```diff
@@ -0,0 +1,353 @@
+					sb->Append(ClassPathSeparatorStringForWindows);
+					sb->Append(fullName);
+				}
+			}
+			catch (System::IO::DirectoryNotFoundException^)
+			{
+				//Ignore invalid paths.
+			}
+		}
+	}
+	auto newPaths = sb->ToString();
+	int len = newPaths->Length;
+	char* finalPath = new char[len + 1];
+	auto hglobal = System::Runtime::InteropServices::Marshal::StringToHGlobalAnsi(newPaths);
+	memcpy(finalPath, (void*)hglobal, len + 1);
+	System::Runtime::InteropServices::Marshal::FreeHGlobal(hglobal);
+	return finalPath;
+}
+
+JavaVMOption* GetJavaOptions(char *argv[], int& optionCount, int firstOptionOrdinal)
+{
+	JavaVMOption* options = new JavaVMOption[optionCount + 1];
+	char classPathBuf[maxPathBufSize];
+	int sourceOrdinal = firstOptionOrdinal;
+
+	for (int i = 0; i < optionCount; i++) {
+		SetOption(options + i, argv[sourceOrdinal]);
+		if (0 == strcmp(argv[i + firstOptionOrdinal], JavaOptionClassPath) && ((i + 1) < optionCount)) {
+			strcpy_s(classPathBuf, argv[++sourceOrdinal]);
+			for (char* ptr = classPathBuf; *ptr; ptr++) {
+				if (*ptr == '/') {
+					*ptr = '\\';
+				}
+			}
+			strcat_s(classPathBuf, ";local\\*;global\\*");
+			auto expandedPath = ExpandJarPaths(classPathBuf);
+			SetOption(options + i, expandedPath);
+			--optionCount;
+		}
+		++sourceOrdinal;
+	}
+
+	return options;
+}
+
+int Get_CreateJavaVM_Function(JNI_CreateJavaVM_FN& fn_JNI_CreateJavaVM)
+{
+	wchar_t jvmDllPath1[maxPathBufSize];
+	wchar_t jvmDllPath2[maxPathBufSize];
+	DWORD rc = GetEnvironmentVariable(JAVA_HOME, jvmDllPath1, maxPathBufSize);
+	if (0 == rc) {
+		wprintf(L"Could not GetEnvironmentVariable %ls\n", JAVA_HOME);
+		return ErrGetEnvironmentVariable;
+	}
+
+	wcscat_s(jvmDllPath1, maxPathBufSize, JVM_DLL1);
+
+	HMODULE jvm_dll = LoadLibrary(jvmDllPath1);
+	if (jvm_dll == NULL) {
+		wprintf(L"Could not load dll %ls\n", jvmDllPath1);
+		GetEnvironmentVariable(JAVA_HOME, jvmDllPath2, maxPathBufSize);
+		wcscat_s(jvmDllPath2, maxPathBufSize, JVM_DLL2);
+		jvm_dll = LoadLibrary(jvmDllPath2);
+		if (jvm_dll == NULL) {
+			wprintf(L"Could not load dll %ls\n", jvmDllPath2);
+			return ErrLoadLibraryJVM;
+		}
+	}
+
+	fn_JNI_CreateJavaVM = (JNI_CreateJavaVM_FN)GetProcAddress(jvm_dll, JNI_CreateJavaVM_Func_Name);
+	if (fn_JNI_CreateJavaVM == NULL) {
+		printf("Could not GetProcAddress %s\n", JNI_CreateJavaVM_Func_Name);
+		return ErrGetProcAddress;
+	}
+
+	return ErrSuccess;
+}
+
+//
+// Creates Java vm with the given options
+//
+int CreateJVM(JNIEnv*& env, JavaVM*& jvm, JavaVMOption* options, int optionCount) {
+	JNI_CreateJavaVM_FN fn_JNI_CreateJavaVM;
+	int failureCode = 0;
+	if ((failureCode = Get_CreateJavaVM_Function(fn_JNI_CreateJavaVM)) != ErrSuccess) {
+		return failureCode;
+	}
+
+	for (int i = 0; i < optionCount; i++) {
+		printf("Option %d [%s]\n", i, options[i].optionString);
+	}
+	fflush(stdout);
+	JavaVMInitArgs vm_args;
+	memset(&vm_args, 0, sizeof(vm_args));
+	vm_args.version = JNI_VERSION_1_6;
```

Comment:
jni and java versions are somewhat different. If the java version bump does not impact jni, they do not touch it.